packages/@aws-cdk/aws-iam/test/integ.imported-role.ts

```diff
@@ -0,0 +1,17 @@
+import { App, Stack } from '@aws-cdk/core';
+import {
+  IAM_IMPORTED_ROLE_STACK_SAFE_DEFAULT_POLICY_NAME,
+} from '@aws-cdk/cx-api';
+import { PolicyStatement, Role, ServicePrincipal } from '../lib';
+
+const app = new App({ context: { [IAM_IMPORTED_ROLE_STACK_SAFE_DEFAULT_POLICY_NAME]: true } });
+
+const stack = new Stack(app, 'integ-iam-imported-role-1');
+
+const role = new Role(stack, 'TestRole', {
+  assumedBy: new ServicePrincipal('sqs.amazonaws.com'),
+});
+
+role.addToPolicy(new PolicyStatement({ resources: ['*'], actions: ['sqs:SendMessage'] }));
```

Comment:
I don't think this test adds any value. Can you write a test that actually matches the issue?

So likely 3 stacks + assertions
- one to create the role
- two to add different policies
- assertions that the role from Stack 1 has both policies

Reply:
Thanks! Will do!

Reply:
I've added a more precise test, but added the assertion in a separate commit as no other test in iam is using integ tests so far.

Reply:
I'm not surprised about this. The `aws-iam` module and tests are probably some of the older areas of the AWS CDK.